platform/api/equinixmetal/api.go

```diff
@@ -457,28 +457,38 @@ RestartSec=3s
 # Prevent flatcar-install from validating cloud-config
 Environment=PATH=/root/bin:/usr/sbin:/usr/bin
 
-ExecStart=/usr/bin/curl --retry-delay 1 --retry 120 --retry-connrefused --retry-max-time 120 --connect-timeout 20 -fo image.bin.bz2 "%v"
-# We don't verify signatures because the iPXE script isn't verified either
-# (and, in fact, is transferred over HTTP)
-
-ExecStartPre=-/bin/bash -c 'lvchange -an /dev/mapper/*'
-ExecStartPre=-/bin/bash -c 'shopt -s nullglob; for disk in /dev/*d? /dev/nvme?n1; do wipefs --all --force $${disk}; done'
-# 259 is a major number of NVMe devices. They need to be excluded, because
-# the boot agent can't boot from them.
-ExecStart=/usr/bin/flatcar-install -s -e 259 -f image.bin.bz2 %v /userdata
-
-ExecStart=/usr/bin/systemctl --no-block isolate reboot.target
+ExecStart=/opt/installer
 
 StandardOutput=journal+console
 StandardError=journal+console
 
 [Install]
 RequiredBy=multi-user.target
-`, escapedImageURL, userDataOption)
+`
 
 	// make workarounds
 	noopIgnitionConfig := base64.StdEncoding.EncodeToString([]byte(`{"ignition": {"version": "2.1.0"}}`))
 	coreosCloudInit := base64.StdEncoding.EncodeToString([]byte("#!/bin/sh\nexit 0"))
+	installerScript := base64.StdEncoding.EncodeToString([]byte(fmt.Sprintf(`#!/bin/bash
+set -euo pipefail
+curl --retry-delay 1 --retry 120 --retry-connrefused --retry-max-time 120 --connect-timeout 20 -fsSLo image.bin.bz2 "%v"
+# We don't verify signatures because the iPXE script isn't verified either
+# (and, in fact, is transferred over HTTP)
+lvchange -an /dev/mapper/* || true
+shopt -s nullglob
+for disk in /dev/*d? /dev/nvme?n1; do
+  wipefs --all --force "${disk}" || true
+done
+# 259 is a major number of NVMe devices. They need to be excluded, because
+# the boot agent can't boot from them on s3.xlarge.x86.
+INSTANCE=$(curl --retry-delay 1 --retry 120 --retry-connrefused --retry-max-time 120 --connect-timeout 20 -fsSL 'https://metadata.packet.net/metadata' | jq -r '.plan')
+EXCLUDE=""
+if [ "${INSTANCE}" = "s3.xlarge.x86" ]; then
+  EXCLUDE="-e 259"
+fi
+flatcar-install -s ${EXCLUDE} -f image.bin.bz2 %v /userdata
```

Comment:
did you try with another instance?
Just to see if : `flatcar-install -s -f image.bin.bz2 %v /userdata` does not fail? (when `EXCLUDE` is empty)